File: Plugins/engines/org.eclipse.escriptmonkey.scripting.engine.javascript.rhino.samples/samples/scripts/Views__Web_View.js
/*
 * Menu: Examples > Views > JS > Google Web View
 * Kudos: Paul Colton
 * License: EPL 1.0
 * Description : You can use this script to open the Web View.
 * DOM: http://localhost/org.eclipse.eclipsemonkey.lang.javascript
 */

function main()
{
	webView = views.getView("GoogleWebView");
	webView.showView(true);
	webView.setTitle("Google");
	webView.url = "http://www.google.com";
	webView.addEventListener("LocationChanging", onLocationChanging);
}

function onLocationChanging(event)
{
	var location = event.innerEvent.location;
	
	// Print out the location to the Java console
	Packages.java.lang.System.out.println("You clicked on: " + location);
}

File: Plugins/engines/org.eclipse.escriptmonkey.scripting.engine.python.jython.samples/samples/scripts/Views__Web_View.py
#
# Menu: Examples > Views > Google Web View
# Kudos: Paul Colton
# Description: You can use this script to open the Web View.
# License: EPL 1.0
#

from org.eclipse.swt.browser import LocationListener
from org.eclipse.eclipsemonkey.ui.views import GenericScriptableView
from org.eclipse.ui import IWorkbenchPage

class MyLocationListener(LocationListener):
	
	def changing(self, event):
		location = event.location
	
		# Print out the location to the console
		print "You clicked on: " + location
	
	def changed(self, event):
		pass
	
workbench = loadModule("WorkbenchModule")	
page = workbench.getActivePage()

view = page.showView(
	"org.eclipse.eclipsemonkey.ui.scriptableView.GenericScriptableView", 
	"GoogleWebView", 
	IWorkbenchPage.VIEW_VISIBLE)
view.setViewTitle("Google")

browser = view.getBrowser()
browser.setUrl("http://www.google.com")
browser.addLocationListener(MyLocationListener());
